### PR TITLE
Make HumanReadableByteCount serialize to JSON string instead of an object.

### PR DESCRIPTION
**Pull request by GrahamDennis**

### CODE REVIEW

**Comment by GrahamDennis** [comment]
Note: This repo doesn't have a changelog policy, but the changelog status check is required to pass.

File: human-readable-types/src/test/java/com/palantir/humanreadabletypes/HumanReadableByteCountTests.java [hunk 1]
```diff
@@ -19,41 +19,45 @@
...
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import org.junit.Test;
 
 public final class HumanReadableByteCountTests {
+    private static final ObjectMapper objectMapper = new ObjectMapper();
 
     @Test
     public void testParseByte() {
-        assetStringsEqualToBytes(10L, "10", "10b", "10 byte", "10 bytes");
```
Comment (carterkozak) [inline human-readable-types/src/test/java/com/palantir/humanreadabletypes/HumanReadableByteCountTests.java:31]: nice

File: human-readable-types/src/main/java/com/palantir/humanreadabletypes/HumanReadableByteCount.java [hunk 2]
```diff
@@ -245,6 +246,7 @@ public int hashCode() {
      * @return a human-readable string representation of this byte string, not null
      */
     @Override
+    @JsonValue
```
Comment (carterkozak) [inline human-readable-types/src/main/java/com/palantir/humanreadabletypes/HumanReadableByteCount.java:249]: Matches `HumanReadableDuration` which already applies `@JsonValue` to `toString()` :+1:

**Review by carterkozak — APPROVED** [review]

**Comment by iamdanfox** [comment]
One thing that's slightly interesting here is that this type doesn't do a byte-for-byte round-trip, because the jsoncreator does some normalizing.

For example, I think `10 MB` would come out as `10mb`.

I _think_ this is fine, just kinda interesting.

```java
    @JsonCreator
    public static HumanReadableByteCount valueOf(String byteCount) {
        String lower = byteCount.toLowerCase(Locale.ROOT).trim();

        try {
            Matcher matcher = BYTE_COUNT_PATTERN.matcher(lower);

            Preconditions.checkArgument(matcher.matches(), "Invalid byte string: %s", byteCount);

            long size = Long.parseLong(matcher.group(1));
            String suffix = matcher.group(2);

            if (suffix != null && !SUFFIXES.containsKey(suffix)) {
                throw new IllegalArgumentException("Invalid byte string: " + byteCount + ". Wrong byte unit");
            }

            return new HumanReadableByteCount(size, suffix != null ? SUFFIXES.get(suffix) : ByteUnit.BYTE);

        } catch (NumberFormatException e) {
            String byteError = "Size must be specified as bytes (b), "
                    + "kibibytes (k), mebibytes (m), gibibytes (g), tebibytes (t), or pebibytes(p). "
                    + "E.g. 50b, 100k, or 250m.";

            throw new NumberFormatException(byteError + "\n" + e.getMessage());
        }
    }
```

https://github.com/GrahamDennis/human-readable-types/blob/b5f5a804ac1dd2753ce812a230bd9e0d333d7da7/human-readable-types/src/main/java/com/palantir/humanreadabletypes/HumanReadableByteCount.java#L139-L164

**Pull request merged**